src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala

```diff
@@ -0,0 +1,123 @@
+package org.broadinstitute.dsde.rawls.jobexec
+
+import akka.actor.SupervisorStrategy.Restart
+import akka.actor._
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.dataaccess.{WorkflowDAO, ExecutionServiceDAO, DataSource, SubmissionDAO}
+import org.broadinstitute.dsde.rawls.jobexec.SubmissionMonitor.WorkflowStatusChange
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.concurrent.duration.Duration
+
+/**
+ * Created by dvoet on 6/26/15.
+ */
+object SubmissionMonitor {
+  def props(submission: Submission,
+            submissionDAO: SubmissionDAO,
+            workflowDAO: WorkflowDAO,
+            datasource: DataSource,
+            workflowPollInterval: Duration,
+            submissionPollInterval: Duration,
+            workflowMonitorProps: (ActorRef, Workflow) => Props): Props = {
+    Props(new SubmissionMonitor(submission, submissionDAO, workflowDAO, datasource, workflowPollInterval, submissionPollInterval, workflowMonitorProps))
+  }
+
+  sealed trait SubmissionMonitorMessage
+  case class WorkflowStatusChange(workflow: Workflow) extends SubmissionMonitorMessage
+}
+
+/**
+ * An actor that monitors the status of a submission. Each workflow is monitored by a separate actor instantiated
+ * using workflowMonitorProps. Each of those actors will report back via WorkflowStatusChange messages. The submission
+ * actor will watch the workflow actor for termination and mark associated workflows as unknown if they are not yet done.
+ * When there is no workflow activity this actor will wake up every submissionPollInterval to double check (although
```

Comment:
IMO it makes actors cleaner to look at anyways so if you have an actor with 'states' it makes sense to use it. But I'll go back under my cromwell rock and leave you all alone now :)